src/odfesqlodbc/es_communication.cpp

```diff
@@ -438,9 +444,98 @@ int ESCommunication::ExecDirect(const char* query, const char* fetch_size_) {
+            });
+    }
     return 0;
 }
 
+void ESCommunication::SendCursorQueries(std::string cursor) {
+    try {
+        BlockingQueue< ESResult* > queue;
+        auto future_thread = std::async(std::launch::async, [&]() {
+            DataProcessing(&queue);
+        });
+    
+        while (cursor.size() > 0) {
+            std::shared_ptr< Aws::Http::HttpResponse > response = nullptr;
+            IssueRequest(SQL_ENDPOINT_FORMAT_JDBC,
+                         Aws::Http::HttpMethod::HTTP_POST, ctype, "", response,
+                         "", cursor);
+            if (response == nullptr) {
+                m_error_message =
+                    "Failed to receive response from cursor. "
+                    "Received NULL response.";
+                LogMsg(ES_ERROR, m_error_message.c_str());
+                return;
+            }
+            ESResult* es_result = new ESResult;
+            AwsHttpResponseToString(response, es_result->result_json);
+            GetJsonSchema(*es_result);
+            queue.push(es_result);
+            if (es_result->es_result_doc.has("cursor")) {
+                cursor = es_result->es_result_doc["cursor"].as_string();
+            } else {
+                SendCloseCursorRequest(cursor);
+                cursor = "";
+                queue.push(NULL);
+            }
+        }
+    } catch (std::runtime_error& e) {
+        m_error_message =
+            "Received runtime exception: " + std::string(e.what());
+        LogMsg(ES_ERROR, m_error_message.c_str());
+    }
+    return;
+}
+
+void ESCommunication::DataProcessing(BlockingQueue< ESResult* >* queue) {
+    bool loop = TRUE;
+    while (loop) {
+        try {
+            ESResult* es_result = queue->pop();
+            if (es_result != NULL) {
+                if (!m_result_queue.empty()) {
+                    es_result->column_info =
+                        m_result_queue.front()->column_info;
+                    es_result->command_type =
+                        m_result_queue.front()->command_type;
+                    es_result->num_fields = m_result_queue.front()->num_fields;
+                }
+                if (es_result->es_result_doc.has("cursor")) {
+                    es_result->cursor =
+                        es_result->es_result_doc["cursor"].as_string();
+                } else {
+                    loop = FALSE;
+                }
+                m_result_queue.push(std::unique_ptr< ESResult >(es_result));
+            } else {
+                loop = FALSE;
+            }
+        } catch (std::runtime_error& e) {
+            m_error_message = "Failed to get result. " + std::string(e.what());
+            LogMsg(ES_ERROR, m_error_message.c_str());
+            return;
+        }
+    }
+}
+
+void ESCommunication::SendCloseCursorRequest(std::string cursor) {
+    std::shared_ptr< Aws::Http::HttpResponse > response = nullptr;
+    IssueRequest(SQL_ENDPOINT_CLOSE_CURSOR, Aws::Http::HttpMethod::HTTP_POST,
```

Comment:
Is IssueRequest ours? It seems fairly awkward to use, and seems like it should return the response rather than having to pass it to the method.